tests/Functional/Parameters/ParameterTest.php

```diff
@@ -121,18 +121,83 @@ public function testHeaderParameterRequired(): void
     }
 
     #[DataProvider('provideHeaderValues')]
-    public function testHeaderParameterInteger(string $value, int $expectedStatusCode): void
+    public function testHeaderParameter(string $url, array $headers, int $expectedStatusCode): void
     {
-        self::createClient()->request('GET', 'header_integer', ['headers' => ['Foo' => $value]]);
+        self::createClient()->request('GET', $url, ['headers' => $headers]);
         $this->assertResponseStatusCodeSame($expectedStatusCode);
     }
 
     public static function provideHeaderValues(): iterable
     {
-        yield 'valid integer' => ['3', 200];
-        yield 'too high' => ['6', 422];
-        yield 'too low' => ['0', 422];
-        yield 'invalid integer' => ['string', 422];
+        // header_integer
+        yield 'missing header header_integer' => ['header_integer', [], 200];
+        yield 'valid integer header_integer' => ['header_integer', ['Foo' => '3'], 200];
+        yield 'too high header_integer' => ['header_integer', ['Foo' => '6'], 422];
+        yield 'too low header_integer' => ['header_integer', ['Foo' => '0'], 422];
+        yield 'invalid integer header_integer' => ['header_integer', ['Foo' => 'string'], 422];
+
+        // header_float
+        yield 'missing header header_float' => ['header_float', [], 200];
+        yield 'valid float header_float' => ['header_float', ['Bar' => '3.5'], 200];
+        yield 'valid integer header_float' => ['header_float', ['Bar' => '3'], 200];
+        yield 'too high header_float' => ['header_float', ['Bar' => '600'], 422];
+        yield 'too low header_float' => ['header_float', ['Bar' => '0'], 422];
+        yield 'invalid number header_float' => ['header_float', ['Bar' => 'string'], 422];
+
+        // header_boolean
+        yield 'missing header header_boolean' => ['header_boolean', [], 200];
+        yield 'valid boolean false header_boolean' => ['header_boolean', ['Lorem' => 'false'], 200];
+        yield 'valid boolean true header_boolean' => ['header_boolean', ['Lorem' => 'true'], 200];
+        yield 'valid boolean 0 header_boolean' => ['header_boolean', ['Lorem' => 0], 200];
+        yield 'valid boolean 0 string header_boolean' => ['header_boolean', ['Lorem' => '0'], 200];
+        yield 'valid boolean 1 header_boolean' => ['header_boolean', ['Lorem' => 1], 200];
+        yield 'valid boolean 1 string header_boolean' => ['header_boolean', ['Lorem' => '1'], 200];
+        yield 'invalid boolean header_boolean' => ['header_boolean', ['Lorem' => 'string'], 422];
+
+        // header_null
+        yield 'missing header header_null' => ['header_null', [], 200];
+        yield 'valid null header_null' => ['header_null', ['Ipsum' => null], 200];
+        yield 'valid null string header_null' => ['header_null', ['Ipsum' => 'null'], 200];
+        yield 'invalid null header_null' => ['header_null', ['Ipsum' => 'string'], 422];
```

Comment:
you can keep these tests, not sure if there's any real life use case though :D

Reply:
Removed as part of https://github.com/api-platform/core/pull/7240#discussion_r2168365311